File: Python/Easy/3.py
# Challenge: Calculate the Average of a List of Numbers
# Level: Beginner
# Description: This Python challenge involves creating a program that calculates the average of a list of numbers entered by the user.

# Instructions:
# 1. Ask the user to enter a list of numbers separated by spaces.
# 2. Create a function that takes the list of numbers as an argument.
# 3. Inside the function, use the split() method to separate the numbers and convert them into a list.
# 4. Use a loop to iterate over the list and sum all the numbers.
# 5. Divide the sum of the numbers by the count of numbers in the list to calculate the average.
# 6. Return the average.

# Example:
# List of numbers: 5 10 15 20
# Average: 12.5

# Function to calculate the average of a list of numbers
def calculate_average(numbers):
    # Convert the space-separated numbers to a list of floats
    number_list = [float(num) for num in numbers.split()]
    
    # Calculate the sum of numbers
    total = sum(number_list)
    
    # Calculate the average
    average = total / len(number_list)
    
    return average

# Asking user for input
user_numbers = input("Enter a list of numbers separated by spaces: ")

# Calculating the average
result = calculate_average(user_numbers)

# Displaying the result
print(f"Average: {result}")

File: PHP/Easy/3.php
<?php
// Challenge: Calculate the Factorial of a Number
// Level: Beginner
// Description: This challenge involves creating a PHP function that calculates the factorial of an integer.

// Instructions:
// 1. Create a function called factorial that takes an integer as a parameter.
// 2. Inside the function, initialize a variable result to 1. This variable will store the result of the factorial.
// 3. Use a for loop to iterate from 1 to the entered number.
// 4. In each iteration of the loop, multiply the current value of result by the iteration number.
// 5. At the end of the loop, the result variable will contain the factorial of the number.
// 6. Return the value of result.

// Example:
// Number: 5
// Factorial: 5! = 5 x 4 x 3 x 2 x 1 = 120

// Function to calculate the factorial of a number
function factorial($number) {
    // Initialize a variable to store the factorial result
    $result = 1;

    // Use a for loop to calculate the factorial
    for ($i = 1; $i <= $number; $i++) {
        // Multiply the current result by the iteration number
        $result *= $i;
    }

    // Return the factorial result
    return $result;
}

// Example usage:
$userNumber = readline("Enter an integer: ");
$factorialResult = factorial($userNumber);

echo "Factorial: $userNumber! = $factorialResult\n";
?>
